kube-client/src/api/core_methods.rs

```diff
@@ -31,22 +31,22 @@ where
     ///
     /// This function assumes that the object is expected to always exist, and returns [`Error`] if it does not.
     /// Consider using [`Api::get_opt`] if you need to handle missing objects.
-    pub async fn get(&self, name: &str) -> Result<K> {
-        let mut req = self.request.get(name).map_err(Error::BuildRequest)?;
+    pub async fn get(&self, name: &str, gp: &GetParams) -> Result<K> {
```

Comment:
This is probably going to be the most contentious change in here and it is going to lead to significant downstream churn to correct it. `::get` is one of the most commonly used interfaces and the benefit of actually using the params is not necessarily noticeable for most users.

I think it's probably better to have a `get_with` option that defers to `self.request.get(name, gp)` as you've done here, and leave `get`, `entry` etc with a default unset resourceVersion choice .

Reply:
Agreed. 80% of this PR is changing `get` calls throughout tests and docs. As far as breaking changes go, this is a pretty nasty and involved one. I like your plan though, helps us avoid duplicate a bunch of functions and allows people to opt-in to a more performant query.

I'll see if documentation makes sense for `get`, `get_opt` and `entry`, we should maybe point out that it gets the latest resource version so people know what to expect.

Reply:
I like the way you've done the documentation. Clear, and not much duplication of methods. The main use case is preserved, while power users can figure out what patterns to employ themselves if they need to do some combination of the existing functionality (like `get_opt`) with `GetParams`.